matminer/featurizers/composition/roost.py

```diff
@@ -0,0 +1,569 @@
+import functools
+import os
+import numpy as np
+from matminer.featurizers.base import BaseFeaturizer
+from matminer.utils.data import MatscholarElementData
+
+# NOTE comprhys: is this really the only thing we can do here?
+# won't think not throw an error for someone without Roost installed
+# trying to use the RoostFeaturizer?
+try:
+    import torch
+    from torch.utils.data import Dataset, DataLoader
+    from torch.nn import L1Loss, MSELoss, CrossEntropyLoss
+    from torch.utils.tensorboard import SummaryWriter
+
+    from roost.core import Normalizer
+    from roost.roost.model import Roost
+
+    default_device = (
+        torch.device("cuda") if torch.cuda.is_available() else torch.device("cpu")
+    )
+except ImportError:
+    torch = None
+    Dataset, DataLoader = object, object
+    L1Loss, MSELoss, CrossEntropyLoss = object, object, object
+    SummaryWriter = object
+
+    Normalizer = object
+    Roost = object
+    default_device = None
+
+
+class RoostFeaturizer(BaseFeaturizer):
+    """
+    Representation learning from stiochiometry
+    """
+
+    def __init__(
```

Comment:
Generally, even for the 5k data set I used for developing I saw that the 3-3-256-256 setup for the graph was still locally pretty good and that making the output nn larger boosted performance. I settled on the defaults because anything much larger comes with diminishing returns.